### ISSUE TITLE
Arrays - accessing elements

### ISSUE THREAD

**Issue by dearshrewdwit** [boolean-uk/js-fundamentals#19]
Problem: hardcoding strings, rather than accessing elements from the array.
> https://github.com/boolean-uk/js-fundamentals/pull/9/files#diff-3c128a8cc352a601cff4631181bb3e3f61edba423a93e29992fd84aaa4a99939

**Comment by Auenc** [comment]
Perhaps this is something we just need to keep an eye on and mention to students that they need to achieve this via other methods?

I think it would be rather difficult to try and get the tests/some library to pick this up.

**Comment by dearshrewdwit** [comment]
Agreed - it sounds like a lot of effort for little reward. I wonder if clarifying the written instructions and including examples that demonstrate accessing elements with bracket notation could help reduce the incidence of this.

**Comment by dearshrewdwit** [comment]
There might be an easy win actually to change the array element values to avoid comparing primitive data types -> using String objects, or other objects.

**Comment by Auenc** [comment]
Maybe a custom object? With an implementation something like:
```js
class Location {
  constructor(location) {
    this.location = location
    this.randomNumber = Math.random()
  }
}
```

That way it wouldn't be possible to copy and paste the initialisation if the test checks against the instance within the array. However the test errors might not be the easiest for students to read :thinking:

**Comment by dearshrewdwit** [comment]
Want to create a PR with a quick example?

**Comment by Auenc** [comment]
Here is a link to the PR: https://github.com/boolean-uk/js-fundamentals/pull/22

**Comment by Auenc** [comment]
Should probably note that my linter was screaming at me when writing that, so the solution might not be in the best of shapes due to conflicting rules.
